lib/reqTemplate.js

```diff
@@ -95,6 +96,50 @@ var globalMethods = {
         };
     },
 
+    /**
+     * Formats a date in a requested format
+     *
+     * @param date date to format
+     * @param [format] optional format, defaults to RFC 822 format
+     */
+    date: function(date, format) {
+
+        function isValidDate(d) {
+            if (d.constructor !== Date) {
+                return false;
+            }
+            return !isNaN(d.getTime());
+        }
+
+        if (!isValidDate(date)) {
+            var origDate = date;
+            format = format || 'rfc822';
+
+            if (typeof date === 'string') {
+                // It's a string, but may be it's just a stringified timestamp.
+                // Check if it actually is.
+                var tempNumDate = parseInt(date);
+                if (!Number.isNaN(tempNumDate)) {
+                    date = tempNumDate;
+                }
+            }
+            date = new Date(date);
```

Comment:
It is inside the `if(!isValidDate())`, so if the `date` parameter is already a `Date`, this code will not be executed.